### DISCUSSION TITLE
User Copy Mode and COMMIT_AND_FETCH interaction

### DISCUSSION THREAD

**Discussion by sargun** [ming1/ubdsrv#65] [General]
Just so I understand, you still need to do the COMMIT_AND_FETCH command after the user copy, right?

So, the interaction flow goes something like this:

```
server->driver: FETCH_REQ
driver->server: descriptor for read
server->driver: pwritev($SPECIAL_OFFSET, buf)
server->driver: COMMIT_AND_FETCH(descriptor)
```

**Comment by ming1** [comment]
Yes.

Exactly,  for READ, you need to call pwritev() to write data into the IO buffer.
for WRITE, you need to call preadv() to read data from the IO buffer.

USER_COPY just means the ublk_drv(kernel side) doesn't handle the IO
data copy.

thanks,
